app/producers/application_producer.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Common Kafka producer client
+# https://www.rubydoc.info/gems/ruby-kafka/Kafka/Producer
+class ApplicationProducer < Kafka::Client
+  BROKERS = [ENV['KAFKAMQ']].compact.freeze
+  CLIENT_ID = 'compliance-payload-tracker-producer'
+  SERVICE = 'compliance'
+  # Define TOPIC in the inherited class.
+  # Example:
+  #   TOPIC = 'platform.payload-status'
```

Comment:
The typical way to do this is to use a method instead of a constant and `raise NotImplementedError` in the parent class. This will enforce descendants implement their own custom version of the method:

```rb
def topic
  raise NotImplementedError.new('topic must be defined')
end
```

Reply:
I prefer the use of a constant here. If it would not be defined then it would also error out.